DNN Platform/Library/Common/Lists/ListController.cs

```diff
@@ -1,416 +1,627 @@
-                c => CBO.FillCollection<ListEntryInfo>(DataProvider.Instance().GetListEntriesByListName(listName, string.Empty, portalId)));
-        }
-    }
-}
+    /// <summary>
+    /// Provides access to Dnn Lists.
+    /// </summary>
+    public class ListController
+    {
+        /// <summary>
+        /// The list of list types that are not localized.
+        /// </summary>
+        [Obsolete("Deprecated in v9.8.1, use UnLocalizedLists instead, schedule removal in v11.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "StyleCop.CSharp.MaintainabilityRules",
+            "SA1401:Fields should be private",
+            Justification = "Make private in v11.")]
+        public readonly string[] NonLocalizedLists = UnLocalizableLists;
+
+        private static readonly string[] UnLocalizableLists = { "ContentTypes", "Processor", "DataType", "ProfanityFilter", "BannedPasswords" };
+
+        /// <summary>
+        /// Gets the lists that do not support localization.
+        /// </summary>
+        public string[] UnLocalizedLists => UnLocalizableLists;
+
+        /// <summary>
+        /// Adds a new list entry to the database. If the current thread locale is not "en-US" then the text value will also be
+        /// persisted to a resource file under App_GlobalResources using the list's name and the value as key.
+        /// </summary>
+        /// <param name="listEntry">The list entry.</param>
+        /// <returns>The entry id.</returns>
+        public int AddListEntry(ListEntryInfo listEntry)
+        {
+            bool enableSortOrder = listEntry.SortOrder > 0;
+            this.ClearListCache(listEntry.PortalID);
+            int entryId = DataProvider.Instance().AddListEntry(
+                listEntry.ListName,
+                listEntry.Value,
+                listEntry.TextNonLocalized,
+                listEntry.ParentID,
+                listEntry.Level,
+                enableSortOrder,
+                listEntry.DefinitionID,
+                listEntry.Description,
+                listEntry.PortalID,
+                listEntry.SystemList,
+                UserController.Instance.GetCurrentUserInfo().UserID);
+
+            if (entryId != Null.NullInteger)
+            {
+                // TODO: Refactor this usage as decided in https://github.com/dnnsoftware/Dnn.Platform/issues/4242
+                EventLogController.Instance.AddLog(
+                    listEntry,
+                    PortalController.Instance.GetCurrentSettings(),
+                    UserController.Instance.GetCurrentUserInfo().UserID,
+                    string.Empty,
+#pragma warning disable CS0618 // Type or member is obsolete
+                    EventLogController.EventLogType.LISTENTRY_CREATED);
+#pragma warning restore CS0618 // Type or member is obsolete
+            }
+
+            if (Thread.CurrentThread.CurrentCulture.Name != Localization.SystemLocale && UnLocalizableLists.Contains(listEntry.ListName))

```

Comment:
Did we lose the `!` here?
```suggestion
            if (Thread.CurrentThread.CurrentCulture.Name != Localization.SystemLocale && !UnLocalizableLists.Contains(listEntry.ListName))
```

Reply:
You are correct